ffi/ffi-macros/src/ontology.rs

```diff
@@ -461,6 +477,111 @@ impl From<SlotValue> for CSlotValue {
     }
 }
 
+impl AsRust<SlotValue> for CSlotValue {
+    fn as_rust(&self) -> Fallible<SlotValue> {
+        match self.value_type {
+            SNIPS_SLOT_VALUE_TYPE::SNIPS_SLOT_VALUE_TYPE_CUSTOM => {
+                Ok(SlotValue::Custom(StringValue {
+                    value: create_rust_string_from!(self.value as *const libc::c_char)
+                }))
+            }
+            SNIPS_SLOT_VALUE_TYPE::SNIPS_SLOT_VALUE_TYPE_NUMBER => {
+                let number_value: f64 = unsafe { *(self.value as *const CNumberValue) };
+                Ok(SlotValue::Number(NumberValue { value: number_value }))
+            }
+            SNIPS_SLOT_VALUE_TYPE::SNIPS_SLOT_VALUE_TYPE_ORDINAL => {
+                let ordinal_value: i64 = unsafe { *(self.value as *const COrdinalValue) };
+                Ok(SlotValue::Ordinal(OrdinalValue { value: ordinal_value }))
+            }
+            SNIPS_SLOT_VALUE_TYPE::SNIPS_SLOT_VALUE_TYPE_INSTANTTIME => {
+                let c_instant_time_value = unsafe { &*(self.value as *const CInstantTimeValue) };
+
+                let instant_time_value = InstantTimeValue {
+                    value: create_rust_string_from!(c_instant_time_value.value),
+                    grain: match c_instant_time_value.grain {
+                        SNIPS_GRAIN::SNIPS_GRAIN_YEAR => Grain::Year,
+                        SNIPS_GRAIN::SNIPS_GRAIN_QUARTER => Grain::Quarter,
+                        SNIPS_GRAIN::SNIPS_GRAIN_MONTH => Grain::Month,
+                        SNIPS_GRAIN::SNIPS_GRAIN_WEEK => Grain::Week,
+                        SNIPS_GRAIN::SNIPS_GRAIN_DAY => Grain::Day,
+                        SNIPS_GRAIN::SNIPS_GRAIN_HOUR => Grain::Hour,
+                        SNIPS_GRAIN::SNIPS_GRAIN_MINUTE => Grain::Minute,
+                        SNIPS_GRAIN::SNIPS_GRAIN_SECOND => Grain::Second
+                    },
+                    precision: match c_instant_time_value.precision {
+                        SNIPS_PRECISION::SNIPS_PRECISION_APPROXIMATE => Precision::Approximate,
+                        SNIPS_PRECISION::SNIPS_PRECISION_EXACT => Precision::Exact
+                    },
+                };
```

Comment:
Can't we implement `AsRust` for `CInstantTimeValue` ?

Reply:
Sure thing, it would be cleaner.